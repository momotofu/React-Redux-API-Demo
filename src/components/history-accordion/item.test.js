import React from 'react'
import { shallow, mount, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import jsdom from 'jsdom'

import AccordionItem from './item'

const doc = jsdom.jsdom('<!doctype html><html><body></body></html>')
global.document = doc
global.window = doc.defaultView

var item =
{
  "open": false,
  "itemID": 'a9asf98h98ahsfoi13',
  "buildID": "Tenrox-R1_1235",
  "itemState": "pending",
  "owner": "samy",
  "timeStarted": 1478128607000,
  "percentageComplete": 77.1,
  "metrics": {
  "test": 64,
  "maintainability": 53,
  "security": 64,
  "workmanship": 72,
  "isCompleted": true
  },
  "build": {
    "timeStamp": 1478128607000,
    "isCompleted": true
  },
  "unitTest": {
    "passed": 142,
    "failed": 10,
    "coveredPercentage": 76,
    "testsPassed": 73,
    "isCompleted": true
  },
  "functionalTest": {
    "passed": 142,
    "failed": 10,
    "coveredPercentage": 76,
    "testsPassed": 68,
    "isCompleted": true
  },
  "result": {
    "status": "find_issue"
  }
}

function handleItemClick(id) {
  return
}


describe('AccordionItem component', function() {
  it('Should match snapshot', function() {
    const wrapper = shallow(<AccordionItem
                                key={item.itemID}
                                {...item}
                                onClick={handleItemClick}
                              />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('Allows props to be set', () => {
    const wrapper = mount(<AccordionItem
                              key={item.itemID}
                              {...item}
                              onClick={handleItemClick}
                            />)
    expect(wrapper.props().functionalTest).toEqual({
      "passed": 142,
      "failed": 10,
      "coveredPercentage": 76,
      "testsPassed": 68,
      "isCompleted": true
    })

    wrapper.setProps({ bar: 'foo' })
    expect(wrapper.props().bar).toEqual('foo')
  })

  it('Method referential transparency', () => {
    const wrapper = mount(<AccordionItem
                                key={item.itemID}
                                {...item}
                                onClick={handleItemClick}
                              />)
    expect(wrapper.instance()
      .getCardProgressBar(18, 'red', 'icon-name', 'label'))
      .toEqual(
        <div className='Accordion-item-description-card-metric-row'>
          <span className="Accordion-item-description-card-metric-progress-bar-icon icon-name" />
          <div className='Accordion-item-description-card-metric-progress-bar'>
            <span
              className="Accordion-item-description-card-metric-progress-bar-meter Accordion-item-description-card-metric-progress-bar-meter-red"
              style={{"width":"18%"}}
              data-label="label"
              />
          </div>
        </div>)
  })

  it('Math methods', () => {
    const wrapper = mount(<AccordionItem
                                key={item.itemID}
                                {...item}
                                onClick={handleItemClick}
                              />)
    expect(wrapper.instance()
      .getChartPercentage(101442.9, 1009813, 100, false))
      .toBeLessThanOrEqual(100)
  })
})