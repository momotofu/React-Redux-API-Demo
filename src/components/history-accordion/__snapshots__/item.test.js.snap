exports[`AccordionItem Should match snapshot 1`] = `
<div
  className="Accordion-item">
  <div
    className="Accordion-item-tab "
    onClick={[Function handleItemClick]}>
    <div
      className="Accordion-item-tab-label">
      <div
        className="Accordion-item-tab-icon-container">
        <span
          className="Accordion-item-tab-icon entypo-publish" />
      </div>
      <div
        className="Accordion-item-tab-label-container">
        <div
          className="Accordion-item-tab-label-title">
          Tenrox-R1_1235
        </div>
        <div
          className="Accordion-item-tab-label-sub">
          <span
            className="Accordion-item-tab-label-owner">
            samy
          </span>
           modified on 11/3/2016, 8:16:47 AM
        </div>
      </div>
    </div>
    <div
      className="Accordion-item-tab-status">
      <span />
    </div>
    <div
      className="Accordion-item-tab-metrics">
      <div
        className="Accordion-item-tab-metric">
        <div
          className="Accordion-item-tab-metric-progress-bar ">
          <span
            className="Accordion-item-tab-metric-progress-bar-meter"
            style={
              Object {
                "height": "100%"
              }
            } />
        </div>
      </div>
      <div
        className="Accordion-item-tab-metric">
        <div
          className="Accordion-item-tab-metric-progress-bar ">
          <span
            className="Accordion-item-tab-metric-progress-bar-meter"
            style={
              Object {
                "height": "100%"
              }
            } />
        </div>
      </div>
      <div
        className="Accordion-item-tab-metric">
        <div
          className="Accordion-item-tab-metric-progress-bar ">
          <span
            className="Accordion-item-tab-metric-progress-bar-meter"
            style={
              Object {
                "height": "100%"
              }
            } />
        </div>
      </div>
      <div
        className="Accordion-item-tab-metric">
        <div
          className="Accordion-item-tab-metric-progress-bar ">
          <span
            className="Accordion-item-tab-metric-progress-bar-meter"
            style={
              Object {
                "height": "77.1%"
              }
            } />
        </div>
      </div>
    </div>
    <div
      className="Accordion-item-tab-signifier-container">
      <div
        className="Accordion-item-tab-signifier entypo-down-open " />
    </div>
  </div>
  <div
    className="Accordion-item-description ">
    <div
      className="Accordion-item-description-card">
      <div
        className="Accordion-item-description-card-header">
        <h1>
          Metrics
        </h1>
      </div>
      <div
        className="Accordion-item-description-card-metric-row">
        <span
          className="Accordion-item-description-card-metric-progress-bar-icon entypo-thermometer" />
        <div
          className="Accordion-item-description-card-metric-progress-bar">
          <span
            className="Accordion-item-description-card-metric-progress-bar-meter Accordion-item-description-card-metric-progress-bar-meter-green"
            data-label="test"
            style={
              Object {
                "width": "64%"
              }
            } />
        </div>
      </div>
      <div
        className="Accordion-item-description-card-metric-row">
        <span
          className="Accordion-item-description-card-metric-progress-bar-icon entypo-tools" />
        <div
          className="Accordion-item-description-card-metric-progress-bar">
          <span
            className="Accordion-item-description-card-metric-progress-bar-meter Accordion-item-description-card-metric-progress-bar-meter-yellow"
            data-label="maintainability"
            style={
              Object {
                "width": "53%"
              }
            } />
        </div>
      </div>
      <div
        className="Accordion-item-description-card-metric-row">
        <span
          className="Accordion-item-description-card-metric-progress-bar-icon entypo-lock" />
        <div
          className="Accordion-item-description-card-metric-progress-bar">
          <span
            className="Accordion-item-description-card-metric-progress-bar-meter Accordion-item-description-card-metric-progress-bar-meter-green"
            data-label="security"
            style={
              Object {
                "width": "64%"
              }
            } />
        </div>
      </div>
      <div
        className="Accordion-item-description-card-metric-row">
        <span
          className="Accordion-item-description-card-metric-progress-bar-icon entypo-brush" />
        <div
          className="Accordion-item-description-card-metric-progress-bar">
          <span
            className="Accordion-item-description-card-metric-progress-bar-meter Accordion-item-description-card-metric-progress-bar-meter-green"
            data-label="workmanship"
            style={
              Object {
                "width": "72%"
              }
            } />
        </div>
      </div>
    </div>
    <div
      className="Accordion-item-description-card">
      <div
        className="Accordion-item-description-card-header">
        <h1>
          Build
        </h1>
      </div>
      <div
        className="Accordion-item-description-card-button-container">
        <div
          className="Accordion-item-description-card-button">
          <div
            className="Icon Icon-large Icon-publish" />
        </div>
        <div
          className="Accordion-item-description-card-button">
          <div
            className="Icon Icon-large Icon-bug" />
        </div>
      </div>
      <div>
        <span
          className="Accordion-item-description-card-footer Accordion-item-description-card-footer-label">
          Debug
        </span>
        <span
          className="Accordion-item-description-card-footer Accordion-item-description-card-footer-label">
          Release
        </span>
      </div>
      <div
        className="Accordion-item-description-card-footer">
        <p>
          11/3/2016, 8:16:47 AM
        </p>
      </div>
    </div>
    <div
      className="Accordion-item-description-card">
      <div
        className="Accordion-item-description-card-header">
        <h1>
          Unit Test
        </h1>
      </div>
      <div
        className="Pie-chart-psuedo-container Pie-chart-psuedo-container-green"
        data-pie="93%">
        <svg
          className="Pie-chart-container"
          height="72"
          width="72">
          <circle
            className="Pie-chart"
            cx="36"
            cy="36"
            r="34"
            strokeDasharray="14, 211" />
        </svg>
        <p
          className="Pie-chart-label">
          Tests passed
        </p>
      </div>
      <div
        className="Accordion-item-description-card-metric-row">
        <span
          className="Accordion-item-description-card-metric-progress-bar-icon entypo-network" />
        <div
          className="Accordion-item-description-card-metric-progress-bar">
          <span
            className="Accordion-item-description-card-metric-progress-bar-meter Accordion-item-description-card-metric-progress-bar-meter-green"
            data-label="code covered"
            style={
              Object {
                "width": "76%"
              }
            } />
        </div>
      </div>
    </div>
    <div
      className="Accordion-item-description-card">
      <div
        className="Accordion-item-description-card-header">
        <h1>
          Functional Test
        </h1>
      </div>
      <div
        className="Pie-chart-psuedo-container Pie-chart-psuedo-container-green"
        data-pie="93%">
        <svg
          className="Pie-chart-container"
          height="72"
          width="72">
          <circle
            className="Pie-chart"
            cx="36"
            cy="36"
            r="34"
            strokeDasharray="14, 211" />
        </svg>
        <p
          className="Pie-chart-label">
          Tests passed
        </p>
      </div>
      <div
        className="Accordion-item-description-card-metric-row">
        <span
          className="Accordion-item-description-card-metric-progress-bar-icon entypo-network" />
        <div
          className="Accordion-item-description-card-metric-progress-bar">
          <span
            className="Accordion-item-description-card-metric-progress-bar-meter Accordion-item-description-card-metric-progress-bar-meter-green"
            data-label="code covered"
            style={
              Object {
                "width": "76%"
              }
            } />
        </div>
      </div>
    </div>
  </div>
</div>
`;
